### ISSUE TITLE
ng xi18n throws exception 'ERROR in Cannot read property 'loadChildren' of undefined' when route data contains an arrow expression

### ISSUE THREAD

**Issue by rsivan** [angular/angular#37776]
# 🐞 bug report

### Is this a regression?
Reported on 10.0.0.
This also happens in 9.x

### Description

ng xi18n reports the following:
 `ERROR in Cannot read property 'loadChildren' of undefined`

## 🔬 Minimal Reproduction
https://github.com/rsivan/route-error

## 🔥 Exception or Error
<pre><code>

> ng xi18n route-stuff --progress --output-path src/locale

ERROR in Cannot read property 'loadChildren' of undefined
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! route-stuff@0.0.0 xi18n: `ng xi18n route-stuff --progress --output-path src/locale`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the route-stuff@0.0.0 xi18n script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:

</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>

Angular CLI: 10.0.0
Node: 13.12.0
OS: darwin x64

Angular: 10.0.0
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.1000.0
@angular-devkit/build-angular     0.1000.0
@angular-devkit/build-optimizer   0.1000.0
@angular-devkit/build-webpack     0.1000.0
@angular-devkit/core              10.0.0
@angular-devkit/schematics        10.0.0
@ngtools/webpack                  10.0.0
@schematics/angular               10.0.0
@schematics/update                0.1000.0
rxjs                              6.5.5
typescript                        3.9.5
webpack                           4.43.0
</code></pre>

**Anything else relevant?**

The root cause appears to be using an arrow expression in a route definition.
Note that the application compiles and runs correctly, just xi18n fails.

`  { path: 'view1', component: View1Component, data: { permission: (r: any) => r.guide > 0 }},`

removing the arrow expression fixes the issue.

**Comment by petebacondarwin** [comment]
Hopefully we will release `10.1.0-next.0` today, which will contain the new i18n message extractor code. Once upgrading you will be able to test this out by running the `localize-extract` binary available in `@angular/localize`. This tool should not have the problem described here, but it is new and may have issues of its own. So it would be great to get some early feedback. We plan to integrate this into the CLI for 10.1.0.

**Comment by petebacondarwin** [comment]
I tried out your reproduction and got it working with the new extraction tooling. This is the diff:

```diff
diff --git a/package.json b/package.json
index d79d3e2..1a329f2 100644
--- a/package.json
+++ b/package.json
@@ -8,15 +8,18 @@
     "test": "ng test",
     "lint": "ng lint",
     "e2e": "ng e2e",
-    "xi18n": "ng xi18n route-stuff --progress --output-path src/locale"
+    "xi18n": "ng xi18n route-stuff --progress --output-path src/locale",
+    "preextract": "ngc -p tsconfig.app.json",
+    "extract": "localize-extract --source \"out-tsc/**/*.js\" --format xliff --outputPath src/locale/messages-2.xlf"
   },
   "private": true,
   "dependencies": {
     "@angular/animations": "~10.0.0",
     "@angular/common": "~10.0.0",
-    "@angular/compiler": "~10.0.0",
+    "@angular/compiler": "https://758735-24195339-gh.circle-artifacts.com/0/angular/compiler-master-289a96b078.tgz",
     "@angular/core": "~10.0.0",
     "@angular/forms": "~10.0.0",
+    "@angular/localize": "https://758735-24195339-gh.circle-artifacts.com/0/angular/localize-master-289a96b078.tgz",
     "@angular/platform-browser": "~10.0.0",
     "@angular/platform-browser-dynamic": "~10.0.0",
     "@angular/router": "~10.0.0",
@@ -27,7 +30,7 @@
   "devDependencies": {
     "@angular-devkit/build-angular": "~0.1000.0",
     "@angular/cli": "~10.0.0",
-    "@angular/compiler-cli": "~10.0.0",
+    "@angular/compiler-cli": "https://758735-24195339-gh.circle-artifacts.com/0/angular/compiler-cli-master-289a96b078.tgz",
     "@types/node": "^12.11.1",
     "@types/jasmine": "~3.5.0",
     "@types/jasminewd2": "~2.0.3",
```

**Comment by rsivan** [comment]
I updated angular and localize to 10.1.0-next.0 but I am not sure how to run the new extraction tool. Can you show me?
Thanks.

**Comment by petebacondarwin** [comment]
My current recommendation is to run two commands. Assuming you have `yarn` installed.

```bash
yarn ngc -p tsconfig.app.json
yarn localize-extract --source "out-tsc/**/*.js" --format xliff --outputPath src/locale/messages-2.xlf
 ```

This assumes that your application is built via the tsconfig.app.json file, and that this will output to the `out-tsc` directory. This depends upon your project setup.

**Comment by rsivan** [comment]
Do you have an npm equivalent?

**Comment by petebacondarwin** [comment]
Sure. These binaries are just stored in `node_modules/.bin` so you could try:

```
node_modules/.bin ngc -p tsconfig.app.json
node_modules/.bin localize-extract --source "out-tsc/**/*.js" --format xliff --outputPath src/locale/messages-2.xlf
```

**Comment by rsivan** [comment]
Verified fixed in 10.1.0-rc.0 using `ng xi18n route-stuff --progress --output-path src/locale --ivy`
